tachyon/math/base/simd_int.h

```diff
@@ -0,0 +1,123 @@
+
+  SimdInt operator&(const SimdInt& other) const;
+  SimdInt& operator&=(const SimdInt& other) { return *this = *this & other; }
+
+  SimdInt operator|(const SimdInt& other) const;
+  SimdInt& operator|=(const SimdInt& other) { return *this = *this | other; }
+
+  SimdInt operator^(const SimdInt& other) const;
+  SimdInt& operator^=(const SimdInt& other) { return *this = *this ^ other; }
+
+  SimdInt operator!() const { return *this ^ Max(); }
+
+  SimdInt operator>>(uint32_t count) const;
+  SimdInt& operator>>=(uint32_t count) { return *this = *this >> count; }
+
+  SimdInt operator<<(uint32_t count) const;
+  SimdInt& operator<<=(uint32_t count) { return *this = *this << count; }
+
+  std::string ToString() const { return value_.ToString(); }
+  std::string ToHexString(bool pad_zero = false) const {
+    return value_.ToHexString(pad_zero);
+  }
+
+ private:
+  value_type value_;
+};
+
+#define SPECIALIZE_SIMD_INT(bits)                                            \
+  using SimdInt##bits = SimdInt<bits>;                                       \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::Broadcast(uint8_t value);                     \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::Broadcast(uint16_t value);                    \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::Broadcast(uint32_t value);                    \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::Broadcast(uint64_t value);                    \
+                                                                             \
+  template <>                                                                \
+  bool SimdInt##bits::operator==(const SimdInt##bits & value) const;         \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::operator&(const SimdInt##bits & value) const; \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::operator|(const SimdInt##bits & value) const; \
+                                                                             \
+  template <>                                                                \
+  SimdInt##bits SimdInt##bits::operator^(const SimdInt##bits & value) const; \
```

Comment:
& fix other related formatting
```suggestion
  bool SimdInt##bits::operator==(const SimdInt##bits& value) const;         \
                                                                            \
  template <>                                                               \
  SimdInt##bits SimdInt##bits::operator&(const SimdInt##bits& value) const; \
                                                                            \
  template <>                                                               \
  SimdInt##bits SimdInt##bits::operator|(const SimdInt##bits& value) const; \
                                                                            \
  template <>                                                               \
  SimdInt##bits SimdInt##bits::operator^(const SimdInt##bits& value) const; \